server/backends/memory_metrics_collector/memory_metrics_collector.go

```diff
@@ -32,6 +33,35 @@ func NewMemoryMetricsCollector() (*MemoryMetricsCollector, error) {
 	}, nil
 }
 
+func (m *MemoryMetricsCollector) IncrementCounts(ctx context.Context, key string, counts map[string]int64) error {
```

Comment:
mind adding unit tests for IncrementCounts and IncrementCountsWithExpiry? We don't have any use cases that would directly exercise these new codepaths so it'd be nice to have some coverage

Reply:
Sure, sounds good.